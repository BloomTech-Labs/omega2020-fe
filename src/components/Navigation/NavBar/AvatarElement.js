import React from 'react';
import Box from '@material-ui/core/Box';
import Avatar from '@material-ui/core/Avatar';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Grow from '@material-ui/core/Grow';
import Paper from '@material-ui/core/Paper';
import Popper from '@material-ui/core/Popper';
import MenuList from '@material-ui/core/MenuList';
import { blue } from '@material-ui/core/colors';
import { makeStyles } from '@material-ui/core/styles';
import LoginLogOut from '../MenuItems/LoginLogOut';
import PersonIcon from '@material-ui/icons/Person';

const AvatarElement = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const anchorRef = React.useRef(null);

  const handleToggle = () => {
    setOpen((prevOpen) => !prevOpen);
  };

  const handleClose = (event) => {
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
      return;
    }

    setOpen(false);
  };

  function handleListKeyDown(event) {
    if (event.key === 'Tab') {
      event.preventDefault();
      setOpen(false);
    }
  }

  // return focus to the button when we transitioned from !open -> open
  const prevOpen = React.useRef(open);
  React.useEffect(() => {
    if (prevOpen.current === true && open === false) {
      anchorRef.current.focus();
    }

    prevOpen.current = open;
  }, [open]);

  return (
    <Box>
      <Avatar
        className={classes.color}
        ref={anchorRef}
        aria-controls={open ? 'menu-list-grow' : undefined}
        aria-haspopup='true'
        onClick={handleToggle}
      >
        <PersonIcon />
      </Avatar>
      <Popper
        placement='bottom-end'
        open={open}
        anchorEl={anchorRef.current}
        role={undefined}
        transition
        disablePortal
      >
        {({ TransitionProps, placement }) => (
          <Grow
            {...TransitionProps}
            style={{
              transformOrigin:
                placement === 'bottom-end' ? 'bottom-end' : 'bottom-end',
            }}
          >
            <Paper>
              <ClickAwayListener onClickAway={handleClose}>
                <MenuList
                  autoFocusItem={open}
                  id='menu-list-grow'
                  onKeyDown={handleListKeyDown}
                >
                  <LoginLogOut handleClose={handleClose} />
                </MenuList>
              </ClickAwayListener>
            </Paper>
          </Grow>
        )}
      </Popper>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'flex-end',
    alignContent: 'flex-end',
    background: '#FFFFFF',
  },
  avatar: {
    marginRight: '10px',
    marginLeft: '25px',
  },
  paper: {
    marginRight: theme.spacing(3),
  },
  MenuItem: {
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    paddingRight: 120,
  },
  color: {
    color: theme.palette.getContrastText(blue.A700),
    backgroundColor: blue.A700,
  },
}));

export default AvatarElement;
