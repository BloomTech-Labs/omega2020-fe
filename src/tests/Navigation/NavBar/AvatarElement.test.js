import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Avatar, Box } from '@material-ui/core';
import AvatarElement from '../../../components/Navigation/NavBar/AvatarElement';

configure({ adapter: new Adapter() });


describe('the length of the material ui and match the snapshot of the foot mobile component', () => {

    const wrapper = mount(<AvatarElement />)

    it('snapshot the component', () => {
        const elementShot = renderer.create(<AvatarElement />).toJSON();
        expect(elementShot).toMatchSnapshot();
    })

    it('render the length of the material ui', () => {
        expect(wrapper.find(Avatar).length).toEqual(1);
        expect(wrapper.find(Box).length).toEqual(1);
    })

})