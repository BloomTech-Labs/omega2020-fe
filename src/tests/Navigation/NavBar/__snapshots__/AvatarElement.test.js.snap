// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the length of the material ui and match the snapshot of the foot mobile component snapshot the component 1`] = `
<div
  className="MuiBox-root MuiBox-root-7"
>
  <div
    aria-haspopup="true"
    className="MuiAvatar-root MuiAvatar-circle makeStyles-color-5 MuiAvatar-colorDefault"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
</div>
`;
